snap/component.go

```diff
@@ -110,8 +110,8 @@ func (c *componentPlaceInfo) Filename() string {
 // will be of the form:
 // /snaps/<snap_instance>/components/<snap_revision>/<component_name>
 func (c *componentPlaceInfo) MountDir() string {
-	return filepath.Join(BaseDir(c.snapInstance), "components",
-		c.snapRevision.String(), c.compName)
+	return filepath.Join(BaseDir(c.snapInstance), "components", c.snapRevision.String(),
+		fmt.Sprintf("%s_%s", c.compName, c.compRevision.String()))
```

Comment:
@alfonsosanchezbeato this kind of path doesn't exist in the spec and is a consequence of the change here, they don't look particularly great tbh. We need to discuss what options we have, using "current" for the final target is a deviation from the spec as is but also adds another level that is not very clear we need for components except for this intermediate paths for the mountpoints. We should discuss on this

Reply:
Changed now as discussed